server/src/main/java/io/druid/indexing/overlord/supervisor/VersionedSupervisorSpec.java

```diff
@@ -45,4 +45,31 @@ public String getVersion()
   {
     return version;
   }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    VersionedSupervisorSpec that = (VersionedSupervisorSpec) o;
+
+    if (getSpec() != null ? !getSpec().equals(that.getSpec()) : that.getSpec() != null) {
+      return false;
+    }
+    return getVersion() != null ? getVersion().equals(that.getVersion()) : that.getVersion() == null;
+
+  }
+
+  @Override
+  public int hashCode()
```

Comment:
I usually use the "java.util.Objects (Java 7+)" template for equals/hashCode in IntelliJ; it is more concise. No need to change this but just an FYI for the future.